micro_sam/evaluation/instance_segmentation.py

```diff
@@ -275,7 +275,11 @@ def run_instance_segmentation_inference(
 
         segmenter.initialize(image, image_embeddings)
         masks = segmenter.generate(**generate_kwargs)
-        instances = mask_data_to_segmentation(masks, with_background=True, min_object_size=min_object_size)
+
+        if len(masks) == 0:  # the instance segmentation can have no masks, hence we just save empty labels
+            instances = np.zeros_like(image)
```

Comment:
I think this may not do the right thing. The image may have multiple channels, but we always want this to be 2d.
You can get the correct image shape from the predictor. I think the following should work (but not tested, so you may need to change a few variable names):
```python
shape = tuple(segmenter._predictor.img_size)  # not sure if it's called img_size, maybe the name is different
image = np.zeros(shape, dtype="unit32")
```

Reply:
Makes sense. Fixed it! Thanks for spotting.